test/integration/external/edges/edges_test.go

```diff
@@ -0,0 +1,177 @@
+package edges
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+	"testing"
+	"text/template"
+	"time"
+
+	"github.com/linkerd/linkerd2/testutil"
+)
+
+var TestHelper *testutil.TestHelper
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(m.Run())
+}
+
+// TestEdges requires that there has been traffic recently between linkerd-web
+// and linkerd-controller for edges to have been registered, which is the
+// case when running this test in the context of the other integration tests.
+
+func TestEdges(t *testing.T) {
```

Comment:
Is there a compelling reason to copy this test into the external suite?  As far as I can tell, the only difference between this edges test and the one in the viz suite is that the expected output is different because prometheus scrapes will be coming from the external prometheus.  I'm not sure this is an especially important difference to test.  I'd consider removing this from the external suite.

Reply:
Yeah, that's right, there are minimal differences between this and the `edges` test from viz. The idea here was to still have a test in place that can verify linkerd works well with an external prometheus instance. Prior to this change, we used to re-run all tests in the deep suite with an external prometheus. 

I think re-running the same tests is a bit overkill, I did however still think it'd be good to have at least one test that can check the interaction between the observability stack and an external prometheus instance. Since none of the other tests in the external suite actually interact with viz, or with prometheus, I thought the easiest thing to do would be to duplicate a test from viz.

I'm ok with removing this if it's redundant.

Reply:
That makes sense.  Having an external suite dedicated to external prometheus is a good idea and I agree that running all the viz tests again is probably overkill.  Is there a `linkerd stat` test that we can duplicate (or share?) for the external suite instead of `linkerd edges`?  Stat is more of a central functionality to viz than edges is, so it might be a better choice.

Reply:
Replaced 👍🏻